### PR TITLE
primary key also is not null and unique key

### PR DESCRIPTION
**Pull request by jm-observer**
## Bug Fixes

- primary key also is not null and unique key

### CODE REVIEW

**Comment by billy1624** [comment]
Hey @jm-observer, thanks for contributing!! You got a story to tell? What is the motivation behind this PR?

**Review by billy1624 — APPROVED** [review]
Thanks!! @jm-observer

File: sea-orm-codegen/src/entity/column.rs [hunk 1]
```diff
@@ -268,14 +268,12 @@ impl From<&ColumnDef> for Column {
...
-        let not_null = col_def
-            .get_column_spec()
-            .iter()
-            .any(|spec| matches!(spec, ColumnSpec::NotNull));
-        let unique = col_def
-            .get_column_spec()
-            .iter()
-            .any(|spec| matches!(spec, ColumnSpec::UniqueKey));
+        let not_null = col_def.get_column_spec().iter().any(|spec| {
+            matches!(spec, ColumnSpec::NotNull) || matches!(spec, ColumnSpec::PrimaryKey)
```
Comment (tyt2y3) [inline sea-orm-codegen/src/entity/column.rs:272]: we can use the shorthand:

```rust
matches!(spec, ColumnSpec::NotNull | ColumnSpec::PrimaryKey)
```

**Comment by tyt2y3** [comment]
What problem does this PR solve? Did a discussion happen somewhere which I missed?

**Comment by billy1624** [comment]
I think the code addition is sounded.
- primary key MUST be not null and unique

File: sea-orm-codegen/src/entity/column.rs (hunk 1 above)
Comment (billy1624) [reply sea-orm-codegen/src/entity/column.rs]: Indeed

Comment (billy1624) [reply sea-orm-codegen/src/entity/column.rs]: Refactored. Please check again

**Comment by jm-observer** [comment]
I am very sorry that I have not noticed the notice.

**Comment by billy1624** [comment]
Hey @jm-observer, no worries! Thanks for your time and contribution :)

**Pull request closed without merging**